docs/audit-log-filter-advanced.md

```diff
@@ -0,0 +1,847 @@
+        {
+          "name": "delete",
+          "abort": true
+        }
+      ]
+    }
+  }
+}
+```
+
+### Conditional Blocking
+
+Block events based on field conditions:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "table_access",
+      "event": [
+        {
+          "name": "update",
+          "abort": {
+            "field": {
+              "name": "table_name.str",
+              "value": "critical_table"
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+## Logical Operators
+
+### AND Operations
+
+Require multiple conditions to be true:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "general",
+      "event": [
+        {
+          "name": "status",
+          "log": {
+            "and": [
+              {
+                "field": {
+                  "name": "general_sql_command.str",
+                  "value": "alter_user"
+                }
+              },
+              {
+                "field": {
+                  "name": "general_user.str",
+                  "value": "root"
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### OR Operations
+
+Log if any condition is true:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "table_access",
+      "event": [
+        {
+          "name": "insert",
+          "log": {
+            "or": [
+              {
+                "field": {
+                  "name": "table_name.str",
+                  "value": "users"
+                }
+              },
+              {
+                "field": {
+                  "name": "table_name.str",
+                  "value": "payments"
+                }
+              }
+            ]
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### NOT Operations
+
+Invert condition results:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "general",
+      "event": [
+        {
+          "name": "status",
+          "log": {
+            "not": {
+              "field": {
+                "name": "general_sql_command.str",
+                "value": "select"
+              }
+            }
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+### Complex Logical Combinations
+
+Combine multiple logical operators:
+
+```json
+{
+  "filter": {
+    "class": {
+      "name": "table_access",
+      "event": [
+        {
+          "name": "update",
+          "log": {
+            "and": [
+              {
+                "or": [
+                  {
+                    "field": {
+                      "name": "table_name.str",
+                      "value": "users"
+                    }
+                  },
+                  {
+                    "field": {
+                      "name": "table_name.str",
+                      "value": "accounts"
+                    }
+                  }
+                ]
+              },
+              {
+                "not": {
+                  "field": {
```

Comment:
Likewise, `table_access|xxx` events do not have `general_user` and therefore cannot be filtered out by this field.
So, may be, change to
```json
"field": {
  "name": "table_database.str",
  "value": "primary_db"
}
```